tests/MongoDB.Driver.Examples/GenerateKeyExamples.cs

```diff
@@ -0,0 +1,306 @@
+            Console.WriteLine($"DataKeyId [base64]: {base64DataKeyId}");
+
+            var mongoClient = new MongoClient(connectionString);
+            var collection = mongoClient
+                .GetDatabase(keyVaultNamespace.DatabaseNamespace.DatabaseName)
+                .GetCollection<BsonDocument>(
+                    keyVaultNamespace.CollectionName,
+                    new MongoCollectionSettings
+                    {
+#pragma warning disable CS0618
+                        GuidRepresentation = GuidRepresentation.Standard
+#pragma warning restore CS0618
+                    });
+            var query = Builders<BsonDocument>.Filter.Eq("_id", new BsonBinaryData(dataKeyId, GuidRepresentation.Standard));
+            var doc = collection
+                .Find(query)
+                .Single();
+        }
+
+        [Fact]
+        public void AzureKey()
+        {
+            var kmsProviders = new Dictionary<string, IReadOnlyDictionary<string, object>>();
+
+            var azureTenantId = Environment.GetEnvironmentVariable("FLE_AZURE_TENANT_ID");
+            var azureClientId = Environment.GetEnvironmentVariable("FLE_AZURE_CLIENT_ID");
+            var azureClientSecret = Environment.GetEnvironmentVariable("FLE_AZURE_CLIENT_SECRET");
+            var azureKmsOptions = new Dictionary<string, object>
+            {
+                { "tenantId", azureTenantId },
+                { "clientId", azureClientId },
+                { "clientSecret", azureClientSecret },
+
+            };
+            kmsProviders.Add("azure", azureKmsOptions);
+
+            var connectionString = "mongodb://localhost:27017";
+            var keyVaultNamespace = CollectionNamespace.FromFullName("encryption.__keyVault");
+            var keyVaultMongoClient = new MongoClient(connectionString);
+            var clientEncryptionOptions = new ClientEncryptionOptions(
+                keyVaultClient: keyVaultMongoClient,
+                keyVaultNamespace: keyVaultNamespace,
+                kmsProviders: kmsProviders);
+
+            var clientEncryption = new ClientEncryption(clientEncryptionOptions);
+            var azureDataKeyKeyName = new BsonString("key-name-csfle");
+            var azureDataKeyKeyVaultEndpoint = new BsonString("key-vault-csfle.vault.azure.net"); // typically <azureKeyName>.vault.azure.net
+            var dataKeyOptions = new DataKeyOptions(
+                masterKey: new BsonDocument
+                {
+                    { "keyName", azureDataKeyKeyName },
+                    { "keyVaultEndpoint", azureDataKeyKeyVaultEndpoint }
+                });
+
+            var dataKeyId = clientEncryption.CreateDataKey("azure", dataKeyOptions, CancellationToken.None);
+            Console.WriteLine($"DataKeyId [UUID]: {dataKeyId}");
+            var base64DataKeyId = Convert.ToBase64String(GuidConverter.ToBytes(dataKeyId, GuidRepresentation.Standard));
```

Comment:
@rstam  can you please take a look at here

Reply:
var dataKeyIdBase64 = ....